### PR TITLE
Allow leading zero with exponents; e.g., "0e1".

### PR DESCRIPTION
**Pull request by jgallagher**
Fixes one of the two issues brought up in #198.

### CODE REVIEW

File: Tests/JSONParserTests.swift [hunk 1]
```diff
@@ -158,6 +158,10 @@ class JSONParserTests: XCTestCase {
             ("  0  ", 0),
             ("123", 123),
             ("  -20  ", -20),
+            ("-0", 0),
+            ("0e1", 0),
+            ("-0e20", 0),
+            ("0.1e1", 1),
```
Comment (jeremy-w) [inline Tests/JSONParserTests.swift]: Maybe check with negative exponent and zero exponent, as well?

Comment (jeremy-w) [inline Tests/JSONParserTests.swift]: Also toss in one with uppercase E to check we handle that variant spelling.

Comment (jeremy-w) [inline Tests/JSONParserTests.swift]: And maybe more fun with leading zeroes, too, like `0001E0001` for `10`.

Comment (jgallagher) [inline Tests/JSONParserTests.swift]: 👍 on neg/zero exponent and uppercase `E`. Leading zeros are not allowed in JSON, and we already have a unit test to confirm they're rejected.

Comment (jeremy-w) [inline Tests/JSONParserTests.swift]: Looks like a leading zero is admitted for the exponent, though. The `number` production uses `int` restricted to be zero or non-zero and then whatever digits, but `exp` is just "at least one digit" in its numeric part, so leading zeroes are legit there.

```
      number = [ minus ] int [ frac ] [ exp ]

      decimal-point = %x2E       ; .

      digit1-9 = %x31-39         ; 1-9

      e = %x65 / %x45            ; e E

      exp = e [ minus / plus ] 1*DIGIT

      frac = decimal-point 1*DIGIT

      int = zero / ( digit1-9 *DIGIT )

      minus = %x2D               ; -

      plus = %x2B                ; +

      zero = %x30                ; 0
```

Comment (jgallagher) [inline Tests/JSONParserTests.swift]: 👍 711f3a7

**Comment by jeremy-w** [comment]
:+1: LGTM

**Pull request merged**